test/test_common/receivers.hpp

```diff
@@ -59,28 +59,43 @@ struct recv_int_ec {
 
 } // namespace empty_recv
 
+class base_expect_receiver {
+  std::atomic<bool> called_{false};
+
+  public:
+  ~base_expect_receiver() { CHECK(called_.load()); }
+  base_expect_receiver() : called_(false) {}
+  base_expect_receiver(base_expect_receiver&& other)
+      : called_(other.called_.exchange(true)) {
+  }
+  base_expect_receiver& operator=(base_expect_receiver&& other) {
+    called_.store(other.called_.exchange(true));
+    return *this;
+  }
+  void set_called() {
+    called_.store(true);
+  }
+};
+
 template<class _Env = empty_env>
-class expect_void_receiver {
+class expect_void_receiver : base_expect_receiver {
   _Env env_;
-  bool called_{false};
 
   public:
-  expect_void_receiver() = default;
-  explicit expect_void_receiver(_Env env) : env_(std::move(env)) {}
-  ~expect_void_receiver() { CHECK(called_); }
+  expect_void_receiver(_Env env = _Env{}) : env_(std::move(env)) {}
 
   expect_void_receiver(expect_void_receiver&& other)
-      : env_(std::move(other.env_))
-      , called_(std::exchange(other.called_, true)) {
+      : base_expect_receiver(std::move(other))
+      , env_(std::move(other.env_)) {
   }
   expect_void_receiver& operator=(expect_void_receiver&& other) {
     env_ = std::move(other.env_);
-    called_ = std::exchange(other.called_, true);
+    base_expect_receiver::operator=(std::move(other));
     return *this;
   }
```

Comment:
This could probably just be default-ed now. Same for the others.

Reply:
well deleting the ones in expected_error_receiver sent me on a merry chase.